app/src/main/kotlin/org/cuberite/android/ui/settings/SettingsScreen.kt

```diff
@@ -0,0 +1,263 @@
+            description = stringResource(R.string.settings_startup_explanation),
+            isChecked = startOnStartup,
+            onCheckedChange = onStartupCheck,
+        )
+        extras()
+    }
+}
+
+@Composable
+private fun WebAdmin(
+    username: String,
+    password: String,
+    webAdminUrl: String?,
+    onOpenWebAdmin: () -> Unit,
+    onOpenWebAdminLogin: (DialogType) -> Unit,
+) {
+    val webAdmin = rememberCategory(
+        title = stringResource(R.string.settings_webadmin_heading),
+        description = """${stringResource(R.string.settings_webadmin_explanation)}
+            
+URL:$webAdminUrl""".trimMargin()
+    )
+    Category(data = webAdmin) {
+        DialogItem(
+            title = stringResource(R.string.settings_webadmin_login),
+            onClick = { onOpenWebAdminLogin(DialogType.WebAdminLogin(username, password)) },
+        )
+        DialogItem(
+            title = stringResource(R.string.settings_webadmin_open),
+            onClick = onOpenWebAdmin,
+        )
+    }
+}
+
+@Composable
+private fun UpdateCuberite(
+    onClick: () -> Unit,
+) {
+    val updateCuberite = rememberCategory(
+        title = stringResource(R.string.settings_install_update),
+        description = stringResource(R.string.settings_install_update_explanation),
+        icon = Icons.Rounded.CloudDownload,
+    )
+    Category(data = updateCuberite) {
+        DialogItem(
+            title = stringResource(R.string.settings_install_update),
+            onClick = onClick,
+        )
+    }
+}
+
+@Composable
+private fun LocalInstall(
+    onResult: (state: InstallService.State, uri: Uri?) -> Unit,
+) {
+
+    val installLocal = rememberCategory(
+        title = stringResource(R.string.settings_install_manually),
+        description = stringResource(R.string.settings_install_manually_explanation),
+        icon = Icons.Rounded.Archive,
+    )
+    val binaryPicker = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.GetContent(),
+        onResult = { uri: Uri? -> onResult(PICK_FILE_BINARY, uri) }
+    )
+    val serverPicker = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.GetContent(),
+        onResult = { uri: Uri? -> onResult(PICK_FILE_SERVER, uri) }
+    )
+    Category(data = installLocal) {
+        DialogItem(
+            title = stringResource(R.string.settings_install_select_binary),
+            onClick = { binaryPicker.launch("*/*") },
+        )
+        DialogItem(
+            title = stringResource(R.string.settings_install_select_server),
+            onClick = { serverPicker.launch("*/*") },
+        )
+    }
+}
+
+@Composable
+private fun About(
+    onClick: (DialogType) -> Unit,
+) {
+    val about = rememberCategory(
+        title = stringResource(R.string.settings_info_heading),
+        footerText = stringResource(R.string.settings_info_version, BuildConfig.VERSION_NAME),
```

Comment:
The version label should open the F-Droid page when clicked.

Reply:
Will add this in two days, I don't have pc right now 😔